### ISSUE TITLE
Future of Blob and Conquer

### ISSUE THREAD

**Issue by gsliepen** [perpendicular-dimensions/blobandconquer#12]
Currently the game is in a bad shape due to missing textures/sound/music, but also due to bitrot. Examples of the latter are:

- OpenGL 1 rendering engine. It's outdated in many ways.
- Resolution selection not of this time. It made sense 20 years ago when we had CRTs and switching screen resolutions was the way to get efficient rendering.
- The build system is too simple, it doesn't properly track dependencies, the only way to do a safe rebuild is to `make clean; make`.
- Compiling map files is not part of the build system and needs to be done manually.
- C++ code is from before C++11

There are several ways this game could be improved and rejuvenated. One way would be to keep the structure of the source code the same as much as possible, but gradually improve on the above mentioned points. Another way would be to port it to a game engine like [Godot](https://godotengine.org/).

If you have any ideas or remarks, please add them as comments to this issue.

**Comment by thesourcehim** [comment]
> * OpenGL 1 rendering engine. It's outdated in many ways.
>
What do you have in mind? GLES, Vulkan? Vulkan is preferable I guess.
> * Resolution selection not of this time. It made sense 20 years ago when we had CRTs and switching screen resolutions was the way to get efficient rendering.
>
Like we discussed in my PR, we could remove resolution selection altogether, replacing it with aspect ratio selection maybe
> * The build system is too simple, it doesn't properly track dependencies, the only way to do a safe rebuild is to `make clean; make`.
>
Adding cmake build files is not too difficult
> * Compiling map files is not part of the build system and needs to be done manually.
>
Good point, but I'm not familiar with appropriate tools, I only did partial recompilation of mission18 once.

Regarding porting to other engine - that would be huge amount of work I guess, on the other hand something needs to be done with one annoying physics bug: slopes can kill (I guess the game thinks you're being crushed when you walk under a slope), which is not such a big problem itself, but it creates another problem: if an enemy holding a keycard walks under a slope and kills itself, keycard becomes stuck inside the slope and irretrievable. Funny enough, moving things like vertically moving platforms which should kill you if you get under it don't actually kill you, you just clip through them to their top.

**Comment by Strzegol** [comment]
> There are several ways this game could be improved and rejuvenated. One way would be to keep the structure of the source code the same as much as possible, but gradually improve on the above mentioned points. Another way would be to port it to a game engine like [Godot](https://godotengine.org/).

I don't think that is a good idea; This would means remaking game practicaly from scratch, if you will want to use high-level Godot features. Of course you can only use "low level" features of Godot, but this would probably only add hardware burden and no real improvements (textures anyway have no normal data or similar stuff).

I suggest following priorities:

1.Replace missing data (gfx first, next sound)
2.Find if is way to fast and automatic update code to C+11, If not, make only sure that proper compiler flags are used (so it don't complain about outdates code, as far as I know GCC can be configured to "understand" old fashion C++).
3.Only fix stuff so average user can play comfortably on modern hardware
4.Only after 1,2,3 completed ypu can think about adding new features.
5.Suggested new feautres:adding normal textures data to nice shadow, fixing obvious game bugs like mentioned slopes. Something alies like to drop off especialy if using a stairs.
